app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -598,5 +598,10 @@ protected function _prepareCustomerQuote($quote)
         if ($shipping && !$shipping->getCustomerId() && !$hasDefaultBilling) {
             $shipping->setIsDefaultBilling(true);
         }
+
+        if (!$billing->getCustomerAddressId() && $shippingAddress) {
```

Comment:
The `$shippingAddress` variable might not be set at this point if all previous conditions are not set. We need to check if this variable is set.